R/library_call_linter.R

```diff
@@ -0,0 +1,60 @@
+#' Library call linter
+#'
+#' Force library calls to all be at the top of the script.
+#'
+#' @examples
+#'  # will produce lints
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')", "library(tidyr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')", "library(tidyr)", "library(purrr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  # okay
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  lint(
+#'    text = c("# comment", "library(dplyr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#' @evalRd rd_tags("library_call_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+library_call_linter <- function() {
+
+  xpath <- "
+  (//SYMBOL_FUNCTION_CALL[text() = 'library'])[last()]
```

Comment:
This can be indented by 2 spaces without changing the logic. That should improve readability a little.

Reply:
Sure thing.  I'll cover this is my next commit!